src/prefect/cli/deploy.py

```diff
@@ -435,7 +440,8 @@ async def _run_single_deploy(
     deploy_config["schedule"] = _construct_schedule(deploy_config, ci=ci)
 
     # determine work pool
-    if get_from_dict(deploy_config, "work_pool.name"):
+    work_pool_name = get_from_dict(deploy_config, "work_pool.name")
```

Comment:
doesn't this line overwrite the default value set above?

Reply:
Before we get here, there's a helper function `_apply_cli_options_to_deploy_config` that merges the provided CLI options and contents from `prefect.yaml`, so if a work pool is not provided via the CLI options or `prefect.yaml`, then `work_pool` will be populated with the value of `PREFECT_DEFAULT_WORK_POOL_NAME`.